tools/shared.py

```diff
@@ -1168,7 +1168,7 @@ def reset(cls):
       # Load the JS defaults into python.
       settings = open(path_from_root('src', 'settings.js')).read().replace('//', '#')
       settings = re.sub(r'var ([\w\d]+)', r'attrs["\1"]', settings)
-      exec(settings, {'attrs': cls.attrs})
+      exec(settings, {'attrs': cls.attrs, 'TARGET_NOT_SUPPORTED': 0x7FFFFFFF})
```

Comment:
Is this needed in addition to the setting in settings.js?

Reply:
Yes, looks like so, otherwise it errors.

Reply:
Ah i see yes because of the above replacement it makes the globals not useable in the definition of other globals.

We could fix the replacement instead but maybe that is actually harder.